src/main.cpp

```diff
@@ -378,19 +385,32 @@ agi::Context& AegisubApp::NewProjectContext() {
 		}
 
 		frames.erase(remove(begin(frames), end(frames), frame), end(frames));
+#ifndef __APPLE__
 		if (frames.empty()) {
 			ExitMainLoop();
 		}
+#endif
 	});
 	frames.push_back(frame);
 	return *frame->context;
 }
 
-void AegisubApp::CloseAll() {
+agi::Context& AegisubApp::EmptyProjectContext() {
+	for (auto& frame : frames) {
+		if (!frame->context->subsController->HasFilename() &&
+				!frame->context->subsController->IsModified())
+			return *frame->context;
```

Comment:
Does this need to also bring the window to the front?

Reply:
I'm actually not seeing new windows being brought to front even before this change, but yes, I should fix that. I'm not sure where the best place to do that is, though.